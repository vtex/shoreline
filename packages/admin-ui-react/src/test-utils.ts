import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
