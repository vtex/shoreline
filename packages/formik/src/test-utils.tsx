import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
