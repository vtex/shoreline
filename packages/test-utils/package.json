{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
