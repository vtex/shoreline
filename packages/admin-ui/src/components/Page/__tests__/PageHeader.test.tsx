import React from 'react'
import { render, axe } from '../../../test-utils'

import { PageHeader, PageTitle, PageActions } from '../index'

describe('PageHeader', () => {
  // TODO The back navigation button is not accessible, it must have an aria-label
  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('should not have any violations', async () => {
    const { container } = render(
      <PageHeader onPopNavigation={() => null}>
        <PageTitle>test paragraph</PageTitle>
        <PageActions>
          <button>Action</button>
        </PageActions>
      </PageHeader>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
