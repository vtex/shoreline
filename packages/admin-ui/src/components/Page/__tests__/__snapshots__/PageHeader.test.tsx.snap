// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: #FFFFFF;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid hsla(0, 0%, 87%, 1);
  position: -webkit-sticky;
  position: sticky;
  top: 0rem;
  height: 4.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  z-index: 999;
}

<header
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`PageHeader should match snapshot with actions 1`] = `
<DocumentFragment>
  .emotion-1 {
  background: #FFFFFF;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid hsla(0, 0%, 87%, 1);
  position: -webkit-sticky;
  position: sticky;
  top: 0rem;
  height: 4.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  z-index: 999;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 >*+* {
  margin-left: 1rem;
}

<header
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      <button>
        Action
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`PageHeader should match snapshot with backlink 1`] = `
<DocumentFragment>
  .emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 *+* {
  margin-left: 1rem;
}

.emotion-4 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 1.25rem;
  line-height: 1.2;
  letter-spacing: -0.0067em;
}

.emotion-6 {
  background: #FFFFFF;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid hsla(0, 0%, 87%, 1);
  position: -webkit-sticky;
  position: sticky;
  top: 0rem;
  height: 4.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  z-index: 999;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: 40px;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(224, 65%, 97%, 1);
}

.emotion-3:active {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

<header
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <button
        class="emotion-3"
        type="button"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          >
            <svg
              class="emotion-0"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="216"
                x2="40"
                y1="128"
                y2="128"
              />
              <polyline
                fill="none"
                points="112 56 40 128 112 200"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
              />
            </svg>
             
          </div>
        </div>
      </button>
      <h1
        class="emotion-4"
      >
        test paragraph
      </h1>
    </div>
  </header>
</DocumentFragment>
`;

exports[`PageHeader should match snapshot with title 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 *+* {
  margin-left: 1rem;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 1.25rem;
  line-height: 1.2;
  letter-spacing: -0.0067em;
}

.emotion-2 {
  background: #FFFFFF;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid hsla(0, 0%, 87%, 1);
  position: -webkit-sticky;
  position: sticky;
  top: 0rem;
  height: 4.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  z-index: 999;
}

<header
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h1
        class="emotion-0"
      >
        test paragraph
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
